### PR TITLE
Node surface tests

### PR DESCRIPTION
**Pull request by nebraszka**

### CODE REVIEW

File: test/src/graph/nodes/SetRingIdsNodeTest.cpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    initializeArgumentsLambda();
+    EXPECT_RGL_INVALID_ARGUMENT(rgl_node_rays_set_ring_ids(&rayNode, ringIds.data(), 0), "ring_ids_count > 0");
+
+    initializeArgumentsLambda();
+    ASSERT_RGL_SUCCESS(rgl_node_rays_from_mat3x4f(&rayNode, rays.data(), rays.size()));
+    ASSERT_RGL_SUCCESS(rgl_node_rays_set_ring_ids(&setRingIdsNode, ringIds.data(), ringIds.size()));
+    ASSERT_RGL_SUCCESS(rgl_graph_node_add_child(rayNode, setRingIdsNode));
+
+    EXPECT_RGL_INVALID_PIPELINE(rgl_graph_run(setRingIdsNode), "ring ids doesn't match number of rays");
```
Comment (msz-rai) [inline test/src/graph/nodes/SetRingIdsNodeTest.cpp:64]: It doesn't match `invalid_arguments` test. I'd create a new one (e.g., `invalid_pipeline`).

File: test/src/graph/nodes/SetRingIdsNodeTest.cpp [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    int32_t idsCount = GetParam();
+    auto ringIds = GenerateRingIds(idsCount);
+    auto rays = GenerateRays(idsCount);
+
+    ASSERT_RGL_SUCCESS(rgl_node_rays_from_mat3x4f(&rayNode, rays.data(), rays.size()));
+    ASSERT_RGL_SUCCESS(rgl_node_rays_set_ring_ids(&setRingIdsNode, ringIds.data(), ringIds.size()));
+    ASSERT_THAT(setRingIdsNode, NotNull());
+    ASSERT_RGL_SUCCESS(rgl_graph_node_add_child(rayNode, setRingIdsNode));
+
+    EXPECT_RGL_SUCCESS(rgl_graph_run(setRingIdsNode));
```
Comment (msz-rai) [inline test/src/graph/nodes/SetRingIdsNodeTest.cpp:79]: Ditto. It should be moved to `valid_pipeline` test.

File: test/src/graph/nodes/YieldPointsNodeTest.cpp [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	yieldNode = nullptr;
+	EXPECT_RGL_INVALID_ARGUMENT(rgl_node_points_yield(&yieldNode, fields, 0), "field_count > 0");
+}
+
+TEST_F(YieldPointsNodeTest, valid_arguments)
+{
+    rgl_node_t yieldNode = nullptr;
+	rgl_field_t fields[] {RGL_FIELD_INTENSITY_F32, RGL_FIELD_AZIMUTH_F32};
+
+	EXPECT_RGL_SUCCESS(rgl_node_points_yield(&yieldNode, fields, 1));
```
Comment (msz-rai) [inline test/src/graph/nodes/YieldPointsNodeTest.cpp:28]: Same as `FormatPointsNodeTests`

File: test/src/meshTest.cpp
Comment (msz-rai) [inline test/src/meshTest.cpp:1]: ?
Also, an empty `sceneTest.cpp` has been created.

**Review by msz-rai — COMMENTED** [review]
Good job!
I have left some comments to address.

File: test/src/graph/nodes/FormatPointsNodeTest.cpp [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	formatNode = nullptr;
+	EXPECT_RGL_INVALID_ARGUMENT(rgl_node_points_format(&formatNode, fields, 0), "field_count > 0");
+}
+
+TEST_F(FormatPointsNodeTests, valid_arguments)
+{
+	rgl_node_t formatNode = nullptr;
+	rgl_field_t fields[] {RGL_FIELD_INTENSITY_F32, RGL_FIELD_AZIMUTH_F32};
+
+	EXPECT_RGL_SUCCESS(rgl_node_points_format(&formatNode, fields, 1));
```
[the remark this answers is not in the document]
Comment (prybicki) [reply test/src/graph/nodes/FormatPointsNodeTest.cpp:28]: There should be a macro to get a size of a local array.
You can also use std::array or std::vector to get sizes automatically.

**Review by prybicki — APPROVED** [review]

**Pull request merged**